frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx

```diff
@@ -117,6 +163,90 @@ describe("QueryBuilder", () => {
     });
   });
 
+  describe("editing models", () => {
+    describe("editing queries", () => {
+      afterEach(() => {
+        jest.resetAllMocks();
+      });
+
+      it("should trigger beforeunload event when leaving edited query", async () => {
+        const { mockEventListener } = await setup({
+          card: TEST_MODEL_CARD,
+          initialRoute: `/model/${TEST_MODEL_CARD.id}/query`,
+        });
+
+        const rowLimitInput = await within(
+          screen.getByTestId("step-limit-0-0"),
```

Comment:
are there multiple rows rendered? I'm curious why do you need to get an element by id first

Reply:
There should only be one row with the query that we're using here - I just wanted to have this here just in case we run into a situation where the row just isn't rendered or something weird happens